File: Leetcode/C++/771. Jewels and Stones/solution.cpp
// https://leetcode.com/problems/jewels-and-stones/

#include <string>
#include <set>

using namespace std;

class Solution
{
public:
    int numJewelsInStones(string jewels, string stones)
    {
        set<char> jewelSet;
        int count = 0;
        for (char jewel : jewels)
        {
            jewelSet.insert(jewel);
        }
        for (char stone : stones)
        {
            if (jewelSet.find(stone) != jewelSet.end())
            {
                count += 1;
            }
        }
        return count;
    }
};

File: Leetcode/Java/771. Jewels and Stones/solution.java
// https://leetcode.com/problems/jewels-and-stones/description/

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> jewelSet = new HashSet<>();
        // Save jewels to jewelSet and find jewel in the stones
        for (char jewel: jewels.toCharArray()){
            jewelSet.add(jewel);
        }

        int count = 0;
        for (char stone: stones.toCharArray()){
            if (jewelSet.contains(stone)){
                count += 1;
            }
        }

        return count;
    }
}